games-puzzle/lpairs/patches/lpairs-1.0.4.patchset

```diff
@@ -69,5 +63,28 @@ index 301878a..a786e39 100644
      else
          strncpy(c_pth, p, sizeof(c_pth)-1);
 -- 
-1.8.3.4
+2.14.2
+
+
+From 51114168f60efd37ff3b7983535512cff43c3123 Mon Sep 17 00:00:00 2001
+From: Shiroko <hhx.xxm@gmail.com>
+Date: Fri, 29 Dec 2017 12:52:29 +0000
+Subject: Solved gcc2->gcc5 problem
+
+
+diff --git a/lpairs/sdl.c b/lpairs/sdl.c
+index 5937097..e4779c4 100644
+--- a/lpairs/sdl.c
++++ b/lpairs/sdl.c
+@@ -480,7 +480,7 @@ void hardware_cap()
+ /*
+     update rectangle (0,0,0,0)->fullscreen
+ */
+-inline void refresh_screen(int x, int y, int w, int h)
++void refresh_screen(int x, int y, int w, int h)
```

Comment:
May I ask what error happened during build that requires you to remove `inline`?

Please also note that `inline` is also specified in `sdl.h`

Reply:
Because the `inline void refresh_screen(int x, int y, int w, int h)` has no definition in header, compiler cannot find the reference to this function. So the `undefined reference to refresh_screen` will happened. The inline function in a cpp file will be only available to that cpp file(it's inline function, so compiler will not export a alone function reference to object). Header  declares but no definition. Another solution is move the body to header(actually this is more reasonable, but I want to keep the minor fix).